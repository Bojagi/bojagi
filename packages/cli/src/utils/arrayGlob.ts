import glob from './glob';

export default async function arrayGlob(
  patterns: string[],
  { executionPath }: { executionPath: string }
) {
  return (
    await Promise.all(patterns.map(storyPath => glob(storyPath, { cwd: executionPath })))
  ).flat();
}
