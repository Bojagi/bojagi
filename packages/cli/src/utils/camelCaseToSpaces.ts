export function camelCaseToSpaces(input: string) {
  return input
    .replace(/([a-z])([A-Z0-9])/g, '$1 $2')
    .replace(/([0-9])([A-Za-z])/g, '$1 $2')
    .split(' ')
    .map(word => word.substr(0, 1).toUpperCase() + word.substr(1))
    .join(' ');
}
