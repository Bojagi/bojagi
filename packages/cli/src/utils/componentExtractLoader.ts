import * as path from 'path';

export function pitch(remainingRequest) {
  // Change the request from an /abolute/path.js to a relative ./path.js
  // This prevents [chunkhash] values from changing when running webpack
  // builds in different directories.
  const newRequestPath = remainingRequest.replace(
    this.resourcePath,
    `.${path.sep}${path.relative(this.context, this.resourcePath)}`
  );
  if (this.cacheable) this.cacheable();
  if (!this.query) throw new Error('query parameter is missing');

  // eslint-disable-next-line no-underscore-dangle
  const moduleName = JSON.stringify(this._module.issuer.name);

  const componentHubFn = 'registerComponent';
  const requirePath = JSON.stringify(`-!${newRequestPath}`);
  const result = `
    var componentModule = require(${requirePath});
    ${componentHubFn}(${moduleName}, componentModule);
    module.exports = componentModule;
  `;

  return result;
}
