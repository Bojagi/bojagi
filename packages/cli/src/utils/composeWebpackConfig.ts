import * as webpack from 'webpack';
import * as pathUtils from 'path';
import { mergeWithCustomize, unique } from 'webpack-merge';

const BLACKLISTED_PLUGINS = ['ManifestPlugin'];

const composeWebpackConfig = (
  baseConfig: webpack.Configuration,
  entry: webpack.Entry,
  executionPath: string,
  decoratorFile: string | undefined,
  getSbOption: <T>(key: string, fallbackValue: T) => T,
  publicPath: string = '__bojagi_public_path__/'
): webpack.Configuration => {
  const { entry: baseEntry, plugins, ...baseConfigWithoutEntry } = baseConfig;
  const filteredPlugins = plugins?.filter(
    plugin => !BLACKLISTED_PLUGINS.includes(plugin.constructor?.name)
  );

  return mergeWithCustomize({
    customizeArray: unique(
      'plugins',
      ['DefinePlugin'],
      plugin => plugin.constructor && plugin.constructor.name
    ),
  })(
    // Bojagi decorator
    {
      module: {
        rules: decoratorFile
          ? [
              {
                test: pathUtils.resolve(executionPath, decoratorFile),
                use: [
                  {
                    loader: `bojagi-expose-loader`,
                    options: {
                      symbol: 'bojagiDecorator',
                    },
                  },
                ],
              },
            ]
          : [],
      },
    },
    {
      ...baseConfigWithoutEntry,
      plugins: filteredPlugins,
    },
    {
      entry,
      output: {
        path: pathUtils.join(process.cwd(), 'bojagi'),
        filename: '[name].js',
        publicPath,
        globalObject: 'window',
      },
      resolve: {
        alias: {
          'storybook-folder': pathUtils.resolve(getSbOption('configDir', './.storybook') || ''),
        },
      },
      resolveLoader: {
        alias: {
          'component-extract-loader': pathUtils.resolve(__dirname, 'componentExtractLoader'),
          'bojagi-expose-loader': pathUtils.resolve(__dirname, 'exposeLoader'),
          reactDom: 'react-dom',
        },
      },
      externals: {
        react: 'react',
        'react-dom': 'reactDom',
      },
      optimization: {
        minimize: true,
        concatenateModules: false,
        splitChunks: {
          cacheGroups: {
            vendors: {
              test: /[\\/]node_modules[\\/]/,
              name: 'vendors',
              chunks: 'all',
            },
          },
        },
      },
      plugins: [
        new webpack.DefinePlugin({
          'process.env': {
            NODE_ENV: JSON.stringify('production'),
          },
        }),
        new webpack.NormalModuleReplacementPlugin(
          /@storybook\/addons/,
          require.resolve('@bojagi/cli/fakeStorybookAddons.js')
        ),
        new webpack.NormalModuleReplacementPlugin(
          /@storybook\/react/,
          require.resolve('@bojagi/cli/fakeStorybookReact.js')
        ),
      ],
    }
  );
};

export default composeWebpackConfig;
