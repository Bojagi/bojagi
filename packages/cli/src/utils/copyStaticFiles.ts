import * as path from 'path';
import { getFS } from '../dependencies';

const fs = getFS();

async function copyStaticFilesFromFolder(folder: string, outputPath: string, subPath: string = '') {
  const folderFiles = await fs.promises.readdir(path.join(folder, subPath));
  return Promise.all(
    folderFiles.map(async folderFile => {
      const relativePath = path.join(subPath, folderFile);
      const fullPath = path.join(folder, relativePath);
      const fullOutputPath = path.join(outputPath, relativePath);
      const stats = await fs.promises.stat(fullPath);

      if (stats.isDirectory()) {
        await fs.promises.mkdir(fullOutputPath);
        return copyStaticFilesFromFolder(folder, outputPath, relativePath);
      }
      await fs.promises.copyFile(fullPath, fullOutputPath);
      return [relativePath];
    })
  ).then(files => files.flat());
}

export async function copyStaticFiles(folders: string[], outputPath: string): Promise<string[]> {
  return Promise.all(
    folders.filter(fs.existsSync).map(folder => copyStaticFilesFromFolder(folder, outputPath))
  ).then(files => files.flat());
}
