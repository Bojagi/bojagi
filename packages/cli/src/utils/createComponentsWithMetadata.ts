import { EntrypointWithMetadata } from '../commands/bundle';
import { Module } from '../types';

export class EntrypointError extends Error {}
export class FilePathError extends Error {}

export default function createComponentsWithMetadata(
  entrypointsWithMetadata: Record<string, EntrypointWithMetadata>,
  compilerOutput: Record<string, string>,
  modules: Module[]
) {
  return Object.entries(entrypointsWithMetadata).reduce((prev, [key, ep]) => {
    const fileContent = compilerOutput[key];
    if (!/^.+?\?.+?\!.+?$/.test(ep.entrypoint)) {
      throw new EntrypointError('Entrypoint does not follow right syntax');
    }

    const fullFilePath = ep.entrypoint.split('!')[1];
    const prefixPath = `${process.cwd()}/`;
    if (!fullFilePath.startsWith(prefixPath)) {
      throw new FilePathError('File does not start with correct path');
    }
    const filePath = fullFilePath.replace(new RegExp(`^${prefixPath}`), '');
    const matchingModule = modules.find(module => module.resource === filePath);
    const dependencies = matchingModule && matchingModule.dependencies;
    const components = ep.components.map(({ symbol, isDefaultExport }) => ({
      fileContent,
      symbol,
      isDefaultExport,
      filePath,
      dependencies,
      exportName: isDefaultExport ? 'default' : symbol,
      name: symbol
    }));
    return [...prev, ...components];
  }, []);
}
