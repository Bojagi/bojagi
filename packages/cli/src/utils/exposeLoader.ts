/*
	MIT License http://www.opensource.org/licenses/mit-license.php
	Author Tobias Koppers @sokra
*/

import * as path from 'path';

function accesorString(value) {
  const childProperties = value.split('.');
  const { length } = childProperties;
  let propertyString = 'global';
  let result = '';

  // eslint-disable-next-line no-plusplus
  for (let i = 0; i < length; i++) {
    if (i > 0) result += `if(!${propertyString}) ${propertyString} = {};\n`;
    propertyString += `[${JSON.stringify(childProperties[i])}]`;
  }

  result += `module.exports = ${propertyString}`;
  return result;
}

// eslint-disable-next-line func-names
export const pitch = function(remainingRequest) {
  // Change the request from an /abolute/path.js to a relative ./path.js
  // This prevents [chunkhash] values from changing when running webpack
  // builds in different directories.
  const newRequestPath = remainingRequest.replace(
    this.resourcePath,
    `.${path.sep}${path.relative(this.context, this.resourcePath)}`
  );

  if (this.cacheable) {
    this.cacheable();
  }

  if (!this.query) throw new Error('query parameter is missing');
  /*
   * Workaround until module.libIdent() in webpack/webpack handles this correctly.
   *
   * fixes:
   * - https://github.com/webpack-contrib/expose-loader/issues/55
   * - https://github.com/webpack-contrib/expose-loader/issues/49
   */
  // eslint-disable-next-line no-underscore-dangle
  const originalUserRequest = this._module.userRequest;
  // eslint-disable-next-line no-underscore-dangle
  this._module.userRequest = `${originalUserRequest}-exposed`;
  const symbol =
    typeof this.query.symbol === 'function'
      ? this.query.symbol(originalUserRequest)
      : this.query.symbol;
  return `${accesorString(symbol)} = require(${JSON.stringify(`-!${newRequestPath}`)});`;
};
