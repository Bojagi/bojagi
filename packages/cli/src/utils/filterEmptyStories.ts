export function filterEmptyStories(story) {
  return !!story.storyItems && story.storyItems.length;
}
