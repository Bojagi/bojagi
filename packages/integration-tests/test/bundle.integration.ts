import { deleteTmpFolder, execBojagi, snapShotTmpFolder } from './utils';

test('run bundle command', () => {
  deleteTmpFolder();
  const result = execBojagi('bundle');
  console.log('result', result);

  snapShotTmpFolder();
});
