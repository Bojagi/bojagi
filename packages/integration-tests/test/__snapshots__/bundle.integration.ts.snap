// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`run bundle command 1`] = `
"{
 \\"version\\": \\"3\\",
 \\"namespaces\\": [
  {
   \\"name\\": \\"default\\",
   \\"framework\\": {
    \\"name\\": \\"react\\",
    \\"version\\": \\"16.12.0\\"
   }
  }
 ]
}"
`;

exports[`run bundle command 2`] = `
"[
 {
  \\"name\\": \\"718ad8b72621629b68f4ea8458266766.png\\",
  \\"namespace\\": \\"default\\",
  \\"outputFilePath\\": \\"files/718ad8b72621629b68f4ea8458266766.png\\"
 },
 {
  \\"name\\": \\"vendors.js\\",
  \\"namespace\\": \\"default\\",
  \\"outputFilePath\\": \\"files/vendors.js\\"
 },
 {
  \\"name\\": \\"App.bojagi.css\\",
  \\"namespace\\": \\"default\\",
  \\"outputFilePath\\": \\"files/App.bojagi.css\\"
 },
 {
  \\"name\\": \\"App.bojagi.js\\",
  \\"namespace\\": \\"default\\",
  \\"outputFilePath\\": \\"files/App.bojagi.js\\"
 },
 {
  \\"name\\": \\"Box.bojagi.js\\",
  \\"namespace\\": \\"default\\",
  \\"outputFilePath\\": \\"files/Box.bojagi.js\\"
 },
 {
  \\"name\\": \\"BoxWithButtons.bojagi.js\\",
  \\"namespace\\": \\"default\\",
  \\"outputFilePath\\": \\"files/BoxWithButtons.bojagi.js\\"
 },
 {
  \\"name\\": \\"Button.bojagi.js\\",
  \\"namespace\\": \\"default\\",
  \\"outputFilePath\\": \\"files/Button.bojagi.js\\"
 },
 {
  \\"name\\": \\"Lazy.bojagi.js\\",
  \\"namespace\\": \\"default\\",
  \\"outputFilePath\\": \\"files/Lazy.bojagi.js\\"
 },
 {
  \\"name\\": \\"Logo.bojagi.js\\",
  \\"namespace\\": \\"default\\",
  \\"outputFilePath\\": \\"files/Logo.bojagi.js\\"
 },
 {
  \\"name\\": \\"PingPong.bojagi.js\\",
  \\"namespace\\": \\"default\\",
  \\"outputFilePath\\": \\"files/PingPong.bojagi.js\\"
 },
 {
  \\"name\\": \\"8.js\\",
  \\"namespace\\": \\"default\\",
  \\"outputFilePath\\": \\"files/8.js\\"
 },
 {
  \\"name\\": \\"./index.html\\",
  \\"namespace\\": \\"default\\",
  \\"outputFilePath\\": \\"files/index.html\\"
 }
]"
`;

exports[`run bundle command 3`] = `
"[
 {
  \\"title\\": \\"App.bojagi\\",
  \\"fileName\\": \\"App.bojagi\\",
  \\"storyItems\\": [
   {
    \\"exportName\\": \\"RenderApp\\",
    \\"storyName\\": \\"Render App\\"
   }
  ],
  \\"dependencies\\": [
   {
    \\"request\\": \\"react\\",
    \\"dependency\\": \\"react\\"
   },
   {
    \\"request\\": \\"./App\\",
    \\"dependency\\": \\"src/App.js\\"
   }
  ],
  \\"gitPath\\": \\"packages/integration-tests/src/App.bojagi.js\\",
  \\"name\\": \\"App.bojagi\\",
  \\"namespace\\": \\"default\\",
  \\"filePath\\": \\"src/App.bojagi.js\\",
  \\"files\\": [
   \\"vendors.js\\",
   \\"App.bojagi.css\\",
   \\"App.bojagi.js\\"
  ]
 },
 {
  \\"title\\": \\"Box.bojagi\\",
  \\"fileName\\": \\"Box.bojagi\\",
  \\"storyItems\\": [
   {
    \\"exportName\\": \\"yellow\\",
    \\"storyName\\": \\"Yellow\\"
   },
   {
    \\"exportName\\": \\"green\\",
    \\"storyName\\": \\"Green\\"
   },
   {
    \\"exportName\\": \\"red\\",
    \\"storyName\\": \\"Red\\"
   },
   {
    \\"exportName\\": \\"blue\\",
    \\"storyName\\": \\"Blue\\"
   },
   {
    \\"exportName\\": \\"brown\\",
    \\"storyName\\": \\"Brown\\"
   }
  ],
  \\"dependencies\\": [
   {
    \\"request\\": \\"react\\",
    \\"dependency\\": \\"react\\"
   },
   {
    \\"request\\": \\"./Box\\",
    \\"dependency\\": \\"src/components/Box.js\\"
   }
  ],
  \\"gitPath\\": \\"packages/integration-tests/src/components/Box.bojagi.js\\",
  \\"name\\": \\"Box.bojagi\\",
  \\"namespace\\": \\"default\\",
  \\"filePath\\": \\"src/components/Box.bojagi.js\\",
  \\"files\\": [
   \\"vendors.js\\",
   \\"Box.bojagi.js\\"
  ]
 },
 {
  \\"title\\": \\"Box With Buttons.bojagi\\",
  \\"fileName\\": \\"BoxWithButtons.bojagi\\",
  \\"storyItems\\": [
   {
    \\"exportName\\": \\"trafficLight\\",
    \\"storyName\\": \\"Traffic Light\\"
   }
  ],
  \\"dependencies\\": [
   {
    \\"request\\": \\"react\\",
    \\"dependency\\": \\"react\\"
   },
   {
    \\"request\\": \\"./BoxWithButtons\\",
    \\"dependency\\": \\"src/components/BoxWithButtons.js\\"
   }
  ],
  \\"gitPath\\": \\"packages/integration-tests/src/components/BoxWithButtons.bojagi.js\\",
  \\"name\\": \\"BoxWithButtons.bojagi\\",
  \\"namespace\\": \\"default\\",
  \\"filePath\\": \\"src/components/BoxWithButtons.bojagi.js\\",
  \\"files\\": [
   \\"vendors.js\\",
   \\"BoxWithButtons.bojagi.js\\"
  ]
 },
 {
  \\"title\\": \\"Button.bojagi\\",
  \\"fileName\\": \\"Button.bojagi\\",
  \\"storyItems\\": [
   {
    \\"exportName\\": \\"buttonYellow\\",
    \\"storyName\\": \\"Button Yellow\\"
   },
   {
    \\"exportName\\": \\"buttonGreen\\",
    \\"storyName\\": \\"Button Green\\"
   }
  ],
  \\"dependencies\\": [
   {
    \\"request\\": \\"react\\",
    \\"dependency\\": \\"react\\"
   },
   {
    \\"request\\": \\"./Button\\",
    \\"dependency\\": \\"src/components/Button.js\\"
   }
  ],
  \\"gitPath\\": \\"packages/integration-tests/src/components/Button.bojagi.js\\",
  \\"name\\": \\"Button.bojagi\\",
  \\"namespace\\": \\"default\\",
  \\"filePath\\": \\"src/components/Button.bojagi.js\\",
  \\"files\\": [
   \\"vendors.js\\",
   \\"Button.bojagi.js\\"
  ]
 },
 {
  \\"title\\": \\"Lazy.bojagi\\",
  \\"fileName\\": \\"Lazy.bojagi\\",
  \\"storyItems\\": [
   {
    \\"exportName\\": \\"lazy\\",
    \\"storyName\\": \\"Lazy\\"
   }
  ],
  \\"dependencies\\": [
   {
    \\"request\\": \\"react\\",
    \\"dependency\\": \\"react\\"
   },
   {
    \\"request\\": \\"./Lazy\\",
    \\"dependency\\": \\"src/components/Lazy.js\\"
   }
  ],
  \\"gitPath\\": \\"packages/integration-tests/src/components/Lazy.bojagi.js\\",
  \\"name\\": \\"Lazy.bojagi\\",
  \\"namespace\\": \\"default\\",
  \\"filePath\\": \\"src/components/Lazy.bojagi.js\\",
  \\"files\\": [
   \\"Lazy.bojagi.js\\"
  ]
 },
 {
  \\"title\\": \\"Logo.bojagi\\",
  \\"fileName\\": \\"Logo.bojagi\\",
  \\"storyItems\\": [
   {
    \\"exportName\\": \\"logo\\",
    \\"storyName\\": \\"Logo\\"
   }
  ],
  \\"dependencies\\": [
   {
    \\"request\\": \\"react\\",
    \\"dependency\\": \\"react\\"
   },
   {
    \\"request\\": \\"./Logo\\",
    \\"dependency\\": \\"src/components/Logo.js\\"
   }
  ],
  \\"gitPath\\": \\"packages/integration-tests/src/components/Logo.bojagi.js\\",
  \\"name\\": \\"Logo.bojagi\\",
  \\"namespace\\": \\"default\\",
  \\"filePath\\": \\"src/components/Logo.bojagi.js\\",
  \\"files\\": [
   \\"Logo.bojagi.js\\"
  ]
 },
 {
  \\"title\\": \\"Ping Pong.bojagi\\",
  \\"fileName\\": \\"PingPong.bojagi\\",
  \\"storyItems\\": [
   {
    \\"exportName\\": \\"PingPong\\",
    \\"storyName\\": \\"Ping Pong\\"
   },
   {
    \\"exportName\\": \\"PongPing\\",
    \\"storyName\\": \\"Pong Ping\\"
   }
  ],
  \\"dependencies\\": [
   {
    \\"request\\": \\"react\\",
    \\"dependency\\": \\"react\\"
   },
   {
    \\"request\\": \\"./Ping\\",
    \\"dependency\\": \\"src/components/Ping.js\\"
   },
   {
    \\"request\\": \\"./Pong\\",
    \\"dependency\\": \\"src/components/Pong.js\\"
   }
  ],
  \\"gitPath\\": \\"packages/integration-tests/src/components/PingPong.bojagi.js\\",
  \\"name\\": \\"PingPong.bojagi\\",
  \\"namespace\\": \\"default\\",
  \\"filePath\\": \\"src/components/PingPong.bojagi.js\\",
  \\"files\\": [
   \\"PingPong.bojagi.js\\"
  ]
 }
]"
`;

exports[`run bundle command 4`] = `
"{
 \\"src/App.bojagi.js\\": {
  \\"isExternal\\": false,
  \\"isNodeModule\\": false,
  \\"filePath\\": \\"src/App.bojagi.js\\",
  \\"gitPath\\": \\"packages/integration-tests/src/App.bojagi.js\\",
  \\"dependencies\\": [
   {
    \\"request\\": \\"react\\",
    \\"dependency\\": \\"react\\"
   },
   {
    \\"request\\": \\"./App\\",
    \\"dependency\\": \\"src/App.js\\"
   }
  ]
 },
 \\"react\\": {
  \\"isExternal\\": true,
  \\"isNodeModule\\": false,
  \\"packageName\\": \\"react\\",
  \\"dependencies\\": []
 },
 \\"src/App.js\\": {
  \\"isExternal\\": false,
  \\"isNodeModule\\": false,
  \\"filePath\\": \\"src/App.js\\",
  \\"gitPath\\": \\"packages/integration-tests/src/App.js\\",
  \\"dependencies\\": [
   {
    \\"request\\": \\"react\\",
    \\"dependency\\": \\"react\\"
   },
   {
    \\"request\\": \\"styled-components\\",
    \\"dependency\\": \\"styled-components\\"
   },
   {
    \\"request\\": \\"./components/Box\\",
    \\"dependency\\": \\"src/components/Box.js\\"
   },
   {
    \\"request\\": \\"./components/Button\\",
    \\"dependency\\": \\"src/components/Button.js\\"
   },
   {
    \\"request\\": \\"./components/BoxWithButtons\\",
    \\"dependency\\": \\"src/components/BoxWithButtons.js\\"
   },
   {
    \\"request\\": \\"./App.css\\",
    \\"dependency\\": \\"src/App.css\\"
   },
   {
    \\"request\\": \\"./components/Lazy\\",
    \\"dependency\\": \\"src/components/Lazy.js\\"
   }
  ]
 },
 \\"styled-components\\": {
  \\"isExternal\\": false,
  \\"isNodeModule\\": true,
  \\"packageName\\": \\"styled-components\\",
  \\"dependencies\\": []
 },
 \\"react-is\\": {
  \\"isExternal\\": false,
  \\"isNodeModule\\": true,
  \\"packageName\\": \\"react-is\\",
  \\"dependencies\\": []
 },
 \\"src/components/Box.js\\": {
  \\"isExternal\\": false,
  \\"isNodeModule\\": false,
  \\"filePath\\": \\"src/components/Box.js\\",
  \\"gitPath\\": \\"packages/integration-tests/src/components/Box.js\\",
  \\"dependencies\\": [
   {
    \\"request\\": \\"react\\",
    \\"dependency\\": \\"react\\"
   },
   {
    \\"request\\": \\"styled-components\\",
    \\"dependency\\": \\"styled-components\\"
   }
  ]
 },
 \\"src/components/Button.js\\": {
  \\"isExternal\\": false,
  \\"isNodeModule\\": false,
  \\"filePath\\": \\"src/components/Button.js\\",
  \\"gitPath\\": \\"packages/integration-tests/src/components/Button.js\\",
  \\"dependencies\\": [
   {
    \\"request\\": \\"react\\",
    \\"dependency\\": \\"react\\"
   },
   {
    \\"request\\": \\"styled-components\\",
    \\"dependency\\": \\"styled-components\\"
   }
  ]
 },
 \\"src/components/BoxWithButtons.js\\": {
  \\"isExternal\\": false,
  \\"isNodeModule\\": false,
  \\"filePath\\": \\"src/components/BoxWithButtons.js\\",
  \\"gitPath\\": \\"packages/integration-tests/src/components/BoxWithButtons.js\\",
  \\"dependencies\\": [
   {
    \\"request\\": \\"react\\",
    \\"dependency\\": \\"react\\"
   },
   {
    \\"request\\": \\"styled-components\\",
    \\"dependency\\": \\"styled-components\\"
   },
   {
    \\"request\\": \\"./Box\\",
    \\"dependency\\": \\"src/components/Box.js\\"
   },
   {
    \\"request\\": \\"./Button\\",
    \\"dependency\\": \\"src/components/Button.js\\"
   }
  ]
 },
 \\"src/App.css\\": {
  \\"isExternal\\": false,
  \\"isNodeModule\\": false,
  \\"filePath\\": \\"src/App.css\\",
  \\"gitPath\\": \\"packages/integration-tests/src/App.css\\",
  \\"dependencies\\": []
 },
 \\"src/components/Lazy.js\\": {
  \\"isExternal\\": false,
  \\"isNodeModule\\": false,
  \\"filePath\\": \\"src/components/Lazy.js\\",
  \\"gitPath\\": \\"packages/integration-tests/src/components/Lazy.js\\",
  \\"dependencies\\": [
   {
    \\"request\\": \\"react\\",
    \\"dependency\\": \\"react\\"
   },
   {
    \\"request\\": \\"styled-components\\",
    \\"dependency\\": \\"styled-components\\"
   }
  ]
 },
 \\"src/components/Box.bojagi.js\\": {
  \\"isExternal\\": false,
  \\"isNodeModule\\": false,
  \\"filePath\\": \\"src/components/Box.bojagi.js\\",
  \\"gitPath\\": \\"packages/integration-tests/src/components/Box.bojagi.js\\",
  \\"dependencies\\": [
   {
    \\"request\\": \\"react\\",
    \\"dependency\\": \\"react\\"
   },
   {
    \\"request\\": \\"./Box\\",
    \\"dependency\\": \\"src/components/Box.js\\"
   }
  ]
 },
 \\"src/components/BoxWithButtons.bojagi.js\\": {
  \\"isExternal\\": false,
  \\"isNodeModule\\": false,
  \\"filePath\\": \\"src/components/BoxWithButtons.bojagi.js\\",
  \\"gitPath\\": \\"packages/integration-tests/src/components/BoxWithButtons.bojagi.js\\",
  \\"dependencies\\": [
   {
    \\"request\\": \\"react\\",
    \\"dependency\\": \\"react\\"
   },
   {
    \\"request\\": \\"./BoxWithButtons\\",
    \\"dependency\\": \\"src/components/BoxWithButtons.js\\"
   }
  ]
 },
 \\"src/components/Button.bojagi.js\\": {
  \\"isExternal\\": false,
  \\"isNodeModule\\": false,
  \\"filePath\\": \\"src/components/Button.bojagi.js\\",
  \\"gitPath\\": \\"packages/integration-tests/src/components/Button.bojagi.js\\",
  \\"dependencies\\": [
   {
    \\"request\\": \\"react\\",
    \\"dependency\\": \\"react\\"
   },
   {
    \\"request\\": \\"./Button\\",
    \\"dependency\\": \\"src/components/Button.js\\"
   }
  ]
 },
 \\"src/components/Lazy.bojagi.js\\": {
  \\"isExternal\\": false,
  \\"isNodeModule\\": false,
  \\"filePath\\": \\"src/components/Lazy.bojagi.js\\",
  \\"gitPath\\": \\"packages/integration-tests/src/components/Lazy.bojagi.js\\",
  \\"dependencies\\": [
   {
    \\"request\\": \\"react\\",
    \\"dependency\\": \\"react\\"
   },
   {
    \\"request\\": \\"./Lazy\\",
    \\"dependency\\": \\"src/components/Lazy.js\\"
   }
  ]
 },
 \\"src/components/Logo.bojagi.js\\": {
  \\"isExternal\\": false,
  \\"isNodeModule\\": false,
  \\"filePath\\": \\"src/components/Logo.bojagi.js\\",
  \\"gitPath\\": \\"packages/integration-tests/src/components/Logo.bojagi.js\\",
  \\"dependencies\\": [
   {
    \\"request\\": \\"react\\",
    \\"dependency\\": \\"react\\"
   },
   {
    \\"request\\": \\"./Logo\\",
    \\"dependency\\": \\"src/components/Logo.js\\"
   }
  ]
 },
 \\"src/components/Logo.js\\": {
  \\"isExternal\\": false,
  \\"isNodeModule\\": false,
  \\"filePath\\": \\"src/components/Logo.js\\",
  \\"gitPath\\": \\"packages/integration-tests/src/components/Logo.js\\",
  \\"dependencies\\": [
   {
    \\"request\\": \\"react\\",
    \\"dependency\\": \\"react\\"
   },
   {
    \\"request\\": \\"./img/logo.png\\",
    \\"dependency\\": \\"src/components/img/logo.png\\"
   }
  ]
 },
 \\"src/components/img/logo.png\\": {
  \\"isExternal\\": false,
  \\"isNodeModule\\": false,
  \\"filePath\\": \\"src/components/img/logo.png\\",
  \\"gitPath\\": \\"packages/integration-tests/src/components/img/logo.png\\",
  \\"dependencies\\": []
 },
 \\"src/components/PingPong.bojagi.js\\": {
  \\"isExternal\\": false,
  \\"isNodeModule\\": false,
  \\"filePath\\": \\"src/components/PingPong.bojagi.js\\",
  \\"gitPath\\": \\"packages/integration-tests/src/components/PingPong.bojagi.js\\",
  \\"dependencies\\": [
   {
    \\"request\\": \\"react\\",
    \\"dependency\\": \\"react\\"
   },
   {
    \\"request\\": \\"./Ping\\",
    \\"dependency\\": \\"src/components/Ping.js\\"
   },
   {
    \\"request\\": \\"./Pong\\",
    \\"dependency\\": \\"src/components/Pong.js\\"
   }
  ]
 },
 \\"src/components/Ping.js\\": {
  \\"isExternal\\": false,
  \\"isNodeModule\\": false,
  \\"filePath\\": \\"src/components/Ping.js\\",
  \\"gitPath\\": \\"packages/integration-tests/src/components/Ping.js\\",
  \\"dependencies\\": [
   {
    \\"request\\": \\"react\\",
    \\"dependency\\": \\"react\\"
   },
   {
    \\"request\\": \\"styled-components\\",
    \\"dependency\\": \\"styled-components\\"
   },
   {
    \\"request\\": \\"./Pong\\",
    \\"dependency\\": \\"src/components/Pong.js\\"
   }
  ]
 },
 \\"src/components/Pong.js\\": {
  \\"isExternal\\": false,
  \\"isNodeModule\\": false,
  \\"filePath\\": \\"src/components/Pong.js\\",
  \\"gitPath\\": \\"packages/integration-tests/src/components/Pong.js\\",
  \\"dependencies\\": [
   {
    \\"request\\": \\"react\\",
    \\"dependency\\": \\"react\\"
   },
   {
    \\"request\\": \\"styled-components\\",
    \\"dependency\\": \\"styled-components\\"
   },
   {
    \\"request\\": \\"./Ping\\",
    \\"dependency\\": \\"src/components/Ping.js\\"
   }
  ]
 }
}"
`;
